src/kaocha/plugin/junit_xml.clj

```diff
@@ -86,21 +86,38 @@
                         ":error")}
      :content [(failure-message m)]}))
 
-(defn testcase->xml [test]
+(defn absolute-path
+  [test-file]
+  (let [abs-path #(.getAbsolutePath %)]
+    (-> test-file
+        (io/resource (deref Compiler/LOADER))
```

Comment:
I don't like directly referencing `Compiler/LOADER` here, as it's depending on an implementation detail IMO. However, omitting it didn't work as expected (`(io/resource test-file) => nil`).  

This works (brittly) because in `kaocha` we can see:
* [`add-classpath`](https://github.com/lambdaisland/kaocha/blob/a505b88ede250135990934e3d1d3b3cf32fede88/src/kaocha/classpath.clj#L37-L51), which is used for adding tests to the classpath, is only ever called with its [single-arity form](https://github.com/lambdaisland/kaocha/search?q=add-classpath)
* The single-arity form of `add-classpath` uses the nullary form of [`classloader-hierarchy`](https://github.com/lambdaisland/kaocha/blob/a505b88ede250135990934e3d1d3b3cf32fede88/src/kaocha/classpath.clj#L18-L24), which in turn binds [`Compiler/LOADER`](https://github.com/lambdaisland/kaocha/blob/a505b88ede250135990934e3d1d3b3cf32fede88/src/kaocha/classpath.clj#L11-L16) to a `DynamicClassLoader` when it's not already bound.
* This will then have the test paths [added](https://github.com/lambdaisland/kaocha/blob/a505b88ede250135990934e3d1d3b3cf32fede88/src/kaocha/classpath.clj#L49), along with any other loaders that have addable classpaths

The situations in which this breaks are:
1. `Compiler/LOADER` is already bound to a classloader that doesn't satisfy [`modifiable-classloader?`](https://github.com/lambdaisland/kaocha/blob/a505b88ede250135990934e3d1d3b3cf32fede88/src/kaocha/classpath.clj#L30), so it never has its classpath added to. I don't know when this would typically happen.
1. `kaocha` updates its internals, which breaks this plugin. Again, I don't know how likely this is, but at least this would be caught by tests when version bumping.

The improvement options I can think of are:
1. Use the same logic as is used for the [classloader determination](https://github.com/lambdaisland/kaocha/blob/a505b88ede250135990934e3d1d3b3cf32fede88/src/kaocha/classpath.clj#L46-L47) to get the loader for this function, perhaps submitting a PR to kaocha to expose this logic. To be fair, it looks like we can just use any classloader that satisfies [`modifiable-classloader?`](https://github.com/lambdaisland/kaocha/blob/a505b88ede250135990934e3d1d3b3cf32fede88/src/kaocha/classpath.clj#L30), to avoid hard-coding a particular one. On the other hand, because of the steps I outlined
2. Do something else completely that I'm missing

Any thoughts welcome here.